test/forked_joins.js

```diff
@@ -0,0 +1,168 @@
+var test = require('tape')
+var hyperlog = require('hyperlog')
+var fdstore = require('fd-chunk-store')
+var path = require('path')
+var memdb = require('memdb')
+var collect = require('collect-stream')
+
+var tmpdir = require('os').tmpdir()
+var storefile = path.join(tmpdir, 'osm-store-' + Math.random())
+
+var osmdb = require('../')
+var versions = {}
+var way0Key
+var osm = osmdb({
+  log: hyperlog(memdb(), { valueEncoding: 'json' }),
+  db: memdb(),
+  store: fdstore(4096, storefile)
+})
+
+test('setup db', function (t) {
```

Comment:
I kind of agree... however, the contained test you wrote is the same as just putting everything in the same file. Having each step as a test is just a cheap way to get async functions to run in order. The advantage is that it enforces order to be top to bottom, whereas writing how you did could end up with the async flow being in different directions if you are not careful. I tend to write tests like this where each `test()` is a step in an async flow, with a setup and teardown step at start and end.